core/src/main/java/io/undertow/util/HttpString.java

```diff
@@ -343,7 +343,7 @@ private static boolean bytesAreEquivalent(final byte[] a, final byte[] b) {
     @SuppressWarnings("deprecation")
     public String toString() {
         if (string == null) {
-            string = new String(bytes, 0);
+            string = new String(bytes, java.nio.charset.Charset.forName("ISO-8859-1"));
```

Comment:
This should be java.nio.charset.StandardCharsets#US_ASCII

Reply:
Got it & thanks!  I will admit I was guessing on the first character translation attempt.  I've pushed a patch (and updated my Updater) with this information.